UnitTests/ScientistTests.cs

```diff
@@ -12,12 +12,41 @@ public class TheScientistClass
     public class TheScienceMethod
     {
         [Fact]
+        public void RunsBothBranchesOfTheExperimentAndMatchesExceptions()
+        {
+            bool candidateRan = false;
+            bool controlRan = false;
+
+            // We introduce side effects for testing. Don't do this in real life please.
+            // Do we do a deep comparison?
+            Func<int> control = () => { controlRan = true; throw new InvalidOperationException(); };
```

Comment:
In the recent unit tests, we use NSubstitute. Would be nice to update this to use that approach. You might need to merge master into this branch (or rebase against master and force push).